pysrc/test_pip_resolve.py

```diff
@@ -1,24 +1,138 @@
 # run with:
-# cd pysrc; python3 pip_resolve_test.py; cd ..
+# cd pysrc; python3 test_pip_resolve.py; cd ..
 
-from pip_resolve import satisfies_python_requirement
+from pip_resolve import satisfies_python_requirement, \
+                        matches_python_version, \
+                        matches_environment
 from collections import namedtuple
 
 import unittest
-
-fake_sys = namedtuple('Sys', ['version_info'])
+from unittest.mock import patch
```

Comment:
I assume we can use python > 3.3 for the tests. For environment testing we should really use `mock` if at all possible.